### PR TITLE
Try reducing reason count and keep MS link, close #1737

### LINKED ISSUE #1737: Still no MS link in long chat messages
#488 was closed last year but the logic in shortening chat messages still sometimes removes the MS link. This seems to annoy and baffle people, probably more than dropping a few more reasons would.

Chat: https://chat.stackexchange.com/transcript/message/43397950#43397950

### PR DESCRIPTION
**Pull request by iBug**
#1737

### CODE REVIEW

File: spamhandling.py [hunk 1]
```diff
@@ -130,9 +127,18 @@ def handle_spam(post, reasons, why):
         log('debug', GlobalVars.parser.unescape(s).encode('ascii', errors='replace'))
         datahandling.append_to_latest_questions(post.post_site, post.post_id, post.title)
 
-        message = prefix_ms + s
+        for reason_count in range(5, 2, -1):  # Try 5 reasons, then 4, then 3
+            reason = ", ".join(reasons[:reason_count])
+            if len(reasons) > reason_count:
+                reason += ", +{} more".format(len(reasons) - reason_count)
+            reason = reason[:1].upper() + reason[1:]  # reason is capitalised, unlike the entries of reasons list
```
Comment (ArtOfCode-) [inline spamhandling.py]: What's this for?

Comment (iBug) [reply spamhandling.py]: Trying to reduce displayed reason count one by one until the message fits.

Comment (ArtOfCode-) [reply spamhandling.py]: Not the loop as a whole - just line 134. What's that do? What's it produce?

Comment (iBug) [reply spamhandling.py]: The first letter would be capitalized by that line.

**Pull request merged**